File: server/assets/solutions/ohtu-avoin-2021/part7/java/KiviPaperiSakset/src/main/java/ohtu/kivipaperisakset/KPSTekoaly.java
package ohtu.kivipaperisakset;

import ohtu.kivipaperisakset.tekoaly.Tekoaly;

public class KPSTekoaly extends KiviPaperiSakset{
    Tekoaly tekoaly;

    public KPSTekoaly(Tekoaly tekoaly) {
        this.tekoaly = tekoaly;
    }
    
    @Override
    protected String toisenSiirto() {
        return tekoaly.annaSiirto();
    }
    
    protected void muistaSiirto(String ekanSiirto) {
        // pelaajan siirto täytyy kertoa tekoälylle
        this.tekoaly.asetaSiirto(ekanSiirto);
    }
}

File: server/assets/solutions/ohtu2023/part7/kivi-paperi-sakset/src/kps_tekoaly.py
from tekoaly import Tekoaly
from kivi_paperi_sakset import KiviPaperiSakset

class KPSTekoaly(KiviPaperiSakset):
    def __init__(self):
            self._tekoaly = Tekoaly()

    def _toisen_siirto(self):
        siirto = self._tekoaly.anna_siirto()
        print("tekoälyn siirto:", siirto)
        return siirto

